strawberry/type.py

```diff
@@ -97,13 +97,17 @@ def _process_type(
     is_interface: bool = False,
     description: Optional[str] = None,
     federation: Optional[FederationTypeParams] = None,
+    private_fields: Optional[List[str]] = None,
 ):
     name = name or to_camel_case(cls.__name__)
 
     wrapped = _wrap_dataclass(cls)
 
     interfaces = _get_interfaces(wrapped)
-    fields = _get_fields(cls)
+    if private_fields:
+        fields = [i for i in _get_fields(cls) if i.origin_name not in private_fields]
+    else:
+        fields = _get_fields(cls)
```

Comment:
Would've written this as
```py
    fields = _get_fields(cls)
    if private_fields:
        fields = list(filter(lambda x: x.origin_name not in private_fields, fields))
```
but mypy was complaining about it
```sh
../../../../../../Users/user/strawberry/strawberry/type:109: error: Unsupported right operand type for in ("Optional[List[str]]") (diff)
```

Reply:
```suggestion
    fields = _get_fields(cls)
    if private_fields:
        fields = [i for i in fields if i.origin_name not in private_fields]
```

Reply:
Not sure why the `list(filter())` doesn't work, but I think this is a bit cleaner

Reply:
Not sure why the list(filter()) doesn't work, but I think this is a bit cleaner